### PR TITLE
bazelisk 1.9.0

### PR DESCRIPTION
**Pull request by philwo**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

**Comment by philwo** [comment]
I would like to get this bottled for arm64, now that I also publish macOS arm64 binaries in the upstream repository.

I know that the test in this formula fails on that platform, but that's because the referenced Bazel version in that test did not yet have arm64 binaries, not because of a problem in Bazelisk, so the failure is expected. The first Bazel version to support macOS arm64 is 4.1.0.

Should we just bump the version number in that test?

**Comment by philwo** [comment]
FYI @chenrui333

**Comment by nandahkrishna** [comment]
> Should we just bump the version number in that test?

Let's give that a go then. Homebrew `bazel` is on `4.0.0` and seems to be bottled for ARM, would that not work?

**Review by carlocab — DISMISSED** [review]
Thanks, @philwo.

**Comment by nandahkrishna** [comment]
@carlocab They still need to change the `bazel` version used in the test.

**Comment by carlocab** [comment]
Restarting the tests so we can see how the ARM runner does. Hopefully this will help us figure out how to fix the test for ARM?

**Comment by carlocab** [comment]
Why can't we just change the targetted version?

**Comment by nandahkrishna** [comment]
We should, as they mentioned (asked in their original comment). I was waiting for @philwo to update it and just slapped the `CI-force-arm` label for any upcoming pushes 😅.

**Comment by carlocab** [comment]
We can make the targetted version conditional on ARM too, I guess, if we want to keep the check for sufficiently old `bazel`s on Intel.

**Comment by philwo** [comment]
Thank you for the comments! I bumped the Bazel version in the test to 4.1.0.

@carlocab If you could let me know how to make the check conditional, I could also implement this. In that case I would recommend to use Bazel 4.0.0 for Intel and 4.1.0 for arm64. But as we're going to very soon release Bazel 4.1.1 or 4.2.0, just using 4.1.0 should also be fine.

File: Formula/bazelisk.rb [hunk 1]
```diff
@@ -39,7 +39,7 @@ def install
     # This is an older than current version, so that we can test that bazelisk
     # will target an explicit version we specify. This version shouldn't need to
     # be bumped.
-    ENV["USE_BAZEL_VERSION"] = "0.28.0"
-    assert_match "Build label: 0.28.0", shell_output("#{bin}/bazelisk version")
+    ENV["USE_BAZEL_VERSION"] = "4.1.0"
+    assert_match "Build label: 4.1.0", shell_output("#{bin}/bazelisk version")
```
Comment (carlocab) [inline Formula/bazelisk.rb:43]:
```suggestion
    bazel_version = Hardware::CPU.arm? ? "4.1.0" : "4.0.0"
    ENV["USE_BAZEL_VERSION"] = bazel_version
    assert_match "Build label: #{bazel_version}", shell_output("#{bin}/bazelisk version")
```
Something like this should work.

**Review by chenrui333 — APPROVED** [review]

**Review by nandahkrishna — APPROVED** [review]
Thanks, @philwo!

**Pull request closed without merging**